docs/source/onnx_dynamo.rst

```diff
@@ -0,0 +1,152 @@
+
+.. code-block:: python
+
+  import torch
+
+  class MNISTModel(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = torch.nn.Conv2d(1, 32, 3, 1, bias=False)
+        self.conv2 = torch.nn.Conv2d(32, 64, 3, 1, bias=False)
+        self.fc1 = torch.nn.Linear(9216, 128, bias=False)
+        self.fc2 = torch.nn.Linear(128, 10, bias=False)
+
+    def forward(self, tensor_x: torch.Tensor):
+        tensor_x = self.conv1(tensor_x)
+        tensor_x = torch.nn.functional.sigmoid(tensor_x)
+        tensor_x = self.conv2(tensor_x)
+        tensor_x = torch.nn.functional.sigmoid(tensor_x)
+        tensor_x = torch.nn.functional.max_pool2d(tensor_x, 2)
+        tensor_x = torch.flatten(tensor_x, 1)
+        tensor_x = self.fc1(tensor_x)
+        tensor_x = torch.nn.functional.sigmoid(tensor_x)
+        tensor_x = self.fc2(tensor_x)
+        output = torch.nn.functional.log_softmax(tensor_x, dim=1)
+        return output
+
+  tensor_x = torch.rand((64, 1, 28, 28), dtype=torch.float32)
+  export_output = torch.onnx.dynamo_export(MNISTModel(), tensor_x)
+
+As the code above shows, all you need is to provide :func:`torch.onnx.dynamo_export` with an instance of the model and its input.
+The exporter will then return an instance of :class:`torch.onnx.ExportOutput` that contains the exported ONNX graph along with extra information.
+
+The in-memory model available through ``export_output.model_proto`` is an ``onnx.ModelProto`` object in compliance with the `ONNX IR spec <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_.
+The ONNX model be serialized into a `Protobuf file <https://protobuf.dev/>`_ using the :meth:`torch.onnx.ExportOutput.save` API.
+
+.. code-block:: python
+
+  export_output.save("/path/to/mnist.onnx")
+
+Inspecting the ONNX model using GUI
+-----------------------------------
+
+You can view the exported model using `Netron <https://netron.app/>`__.
+
+.. image:: _static/img/onnx/onnx_dynamo_mnist_model.png
+    :width: 50%
+    :alt: MNIST model as viewed in Netron
+
+Note that each layer is represented in a rectangular box with a *f* icon in the top right corner.
+
+.. image:: _static/img/onnx/onnx_dynamo_mnist_model_function_highlight.png
+    :width: 50%
+    :alt: ONNX function highlighted
+
+By expanding it, the function body is shown.
+
+.. image:: _static/img/onnx/onnx_dynamo_mnist_model_function_body.png
+    :width: 50%
+    :alt: ONNX function body
+
+The function body is a sequence of ONNX operators or other functions.
+
+Diagnosing issues with SARIF
+----------------------------
+
+ONNX diagnostics goes beyond regular logs through the adoption of
+`Static Analysis Results Interchange Format (aka SARIF) <https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html>`__
+to help users debug and improve their model using a GUI, such as
+Visual Studio Code's `SARIF Viewer <https://marketplace.visualstudio.com/items?itemName=MS-SarifVSCode.sarif-viewer>`_.
```

Comment:
Explain how to open SARIF file after this viewer is installed; e.g.,
- find SARIF file from file exporter window (usually on the right side of VSCode window)
- right click
- ...

SAIRF is a new thing for exporter and even PyTorch. We need to assume users have zero experience with it when writing docs.